examples/gateway.js

```diff
@@ -174,7 +174,10 @@ async function start () {
     gateway: {
       services: [{
         name: 'user',
-        url: 'http://localhost:4001/graphql'
+        url: 'http://localhost:4001/graphql',
+        setResponseHeaders: (reply) => {
```

Comment:
wondering if this would be better as a service-level hook? That would provide us with a lot more flexibility going forward! WDYT? See fastify per route hooks for an example of what I mean: https://www.fastify.io/docs/latest/Reference/Hooks/#route-level-hooks

Reply:
@jonnydgreen ,
If we define it as a service-level hook, we'd need some reconciliation logic in gateway which combines headers sent from multiple subgraphs.
That would be ideal, but I'm not sure where to make that change. Would you guys be able to guide me on where I can write that logic.

Reply:
The idea I had in my was to make it independent of setting headers and leave the reconciliation logic up to the user (which would also fit your use case). E.g. as a pseudo-code example, one could define a service as follows:

```js
{
  name: 'user',
  url: 'http://localhost:4001/graphql',
  onFederatedServiceResponse(gatewayRequest, gatewayReply, serviceDefinition, serviceRequest, serviceReply) {
    // Service specific logic defined by user here
  }
}
```

Looking at the Mercurius codebase, I think this probably requires work that is out of scope of this PR and probably makes sense to do separately (if you're interested, I'd be very happy to guide you with this) so I'm completely fine with your initial approach :)

Reply:
Aces :)